accord-core/src/main/java/accord/coordinate/Exhausted.java

```diff
@@ -32,27 +32,27 @@
 public class Exhausted extends CoordinationFailed
 {
     final Ranges unavailable;
-    private String message;
 
     public Exhausted(TxnId txnId, @Nullable RoutingKey homeKey, Ranges unavailable)
     {
-        super(txnId, homeKey);
+        super(txnId, homeKey, getMessage(txnId, unavailable));
         this.unavailable = unavailable;
     }
 
-    public String getMessage()
-    {
-        if (message == null)
-            message = unavailable == null ? "No more nodes to try" : "No more nodes to try for: " + unavailable;
-        return message;
-    }
-
     Exhausted(TxnId txnId, @Nullable RoutingKey homeKey, Ranges unavailable, Exhausted cause)
     {
         super(txnId, homeKey, cause);
         this.unavailable = unavailable;
     }
 
+    private static String getMessage(TxnId txnId, @Nullable Ranges unavailable)
+    {
+        String msg = "No more nodes to try for " + txnId;
```

Comment:
this isn't a core change, but when debugging you need to know what txn so you can inspect the state.  "something failed" is hard to work with and was hard to debug while creating this patch